src/bsp/platform/stm32f4/xi_bsp_rng_stm32f4.c

```diff
@@ -8,19 +8,30 @@
 
 #if !defined( XI_TLS_LIB_WOLFSSL )
 
+#include "stm32f4xx_hal.h"
+#include "stm32f4xx_hal_rng.h"
+
+RNG_HandleTypeDef xi_stm_rng_handle;
+
 void xi_bsp_rng_init()
 {
+    // The next 2 lines copied from the STMCube-RNG Example
+    /* ##-1- Configure the RNG peripheral ####################################### */
+    xi_stm_rng_handle.Instance = RNG;
+
+    HAL_RNG_Init( &xi_stm_rng_handle );
 }
 
 uint32_t xi_bsp_rng_get()
 {
-    uint32_t random = 0;
-
-    return random;
+    uint32_t random32 = 0;
+    HAL_RNG_GenerateRandomNumber( &xi_stm_rng_handle, &random32 );
```

Comment:
Yes it has. We can't return an error from here, but we should definitely log it.